tests/test_parser.py

```diff
@@ -86,6 +86,21 @@ def test_get_baseimg_from_df(self, dfparser):
         base_img = dfparser.baseimage
         assert base_img.startswith('fedora')
 
+    def test_get_parent_env(self, tmpdir):
+        tmpdir_path = str(tmpdir.realpath())
+        p_env = {"bar": "baz"}
+        df1 = DockerfileParser(tmpdir_path, parent_env=p_env)
+        df1.lines = [
+            "From parent\n",
+            "ENV foo=\"$bar\"\n",
+            "LABEL label=\"$foo $bar\"\n"
+        ]
+
+        assert len(df1.envs) == 1
+        assert df1.envs.get('foo') == 'baz'
+        assert len(df1.labels) == 1
+        assert df1.labels.get('label') == 'baz baz'
```

Comment:
Can we add another test to the end of this which sets df1.lines to ['FROM scratch\n'] and verifies df1.env only has the variables that came from parent_env? I think that would show up the problem with not using .copy() when assigning envs.